pkg/asset/ignition/recovery_ignition.go

```diff
@@ -81,6 +83,28 @@ func (i *RecoveryIgnition) Generate(_ context.Context, dependencies asset.Parent
 		// (even though disabled by default, the udev rule may require it).
 		noConfigImageFile := ignition.FileFromString("/etc/assisted/no-config-image", "root", 0644, "")
 		unconfiguredIgnition.Storage.Files = append(unconfiguredIgnition.Storage.Files, noConfigImageFile)
+
+		ocpVersion := installerConfig.ApplianceConfig.Config.OcpRelease.Version
+		arch := *installerConfig.ApplianceConfig.Config.OcpRelease.CpuArchitecture
+		version := fmt.Sprintf("%s-%s",ocpVersion, arch)
+
+		if installerConfig.ApplianceConfig.Config.OcpRelease.URL != nil && *applianceConfig.Config.OcpRelease.URL != "" {
+			parts := strings.Split(*installerConfig.ApplianceConfig.Config.OcpRelease.URL, ":")
+			if len(parts) >= 2 {
+				version = parts[len(parts)-1]
+			}
+		}
```

Comment:
This sounds like a piece of code that could deserve its own function, ie something like `getCurrentOpenShiftVersion()` ?